config/types.go

```diff
@@ -16,40 +16,34 @@ type Size int64
 
 // UnmarshalTOML parses the duration from the TOML config file
 func (d *Duration) UnmarshalTOML(b []byte) error {
-	var err error
-	b = bytes.Trim(b, `'`)
-
-	// see if we can directly convert it
-	dur, err := time.ParseDuration(string(b))
-	if err == nil {
-		*d = Duration(dur)
-		return nil
-	}
-
-	// Parse string duration, ie, "1s"
-	if uq, err := strconv.Unquote(string(b)); err == nil && len(uq) > 0 {
-		dur, err := time.ParseDuration(uq)
-		if err == nil {
-			*d = Duration(dur)
-			return nil
-		}
-	}
+	// convert to string
+	durStr := string(b)
 
+	// Value is a TOML number (e.g. 3, 10, 3.5)
 	// First try parsing as integer seconds
-	sI, err := strconv.ParseInt(string(b), 10, 64)
+	sI, err := strconv.ParseInt(durStr, 10, 64)
 	if err == nil {
 		dur := time.Second * time.Duration(sI)
 		*d = Duration(dur)
 		return nil
 	}
 	// Second try parsing as float seconds
-	sF, err := strconv.ParseFloat(string(b), 64)
+	sF, err := strconv.ParseFloat(durStr, 64)
 	if err == nil {
 		dur := time.Second * time.Duration(sF)
 		*d = Duration(dur)
 		return nil
 	}
 
+	// Finally, try value is a TOML string (e.g. "3s", 3s) or literal (e.g. '3s')
+	durStr = strings.ReplaceAll(durStr, "'", "")
```

Comment:
Why is this better then using `strconv.Unquote` ?

Reply:
it isn't I forgot about it, I can switch that

Reply:
I didn't realize it would error on non-quoted strings, I assumed it would just return it as-is. Being more explicit sounds like the right approach and not using the function is fine.